proto/buf/alpha/registry/v1alpha1/image.proto

```diff
@@ -34,3 +37,53 @@ message GetImageRequest {
 message GetImageResponse {
   buf.alpha.image.v1.Image image = 1;
 }
+
+enum PartialImageLevel {
+  PARTIAL_IMAGE_LEVEL_UNSPECIFIED = 0;
+
+  // Returns complete files in which the requested types are contained.
+  PARTIAL_IMAGE_LEVEL_FILES = 1;
+
+  // Returns trimmed files in which the requested types are contained. In a
+  // trimmed file, all declarations that are not embedded (directly or
+  // indirectly) in requested types are removed.
+  PARTIAL_IMAGE_LEVEL_TYPES = 2;
+}
+
+message GetPartialImageRequest {
+  string owner = 1;
+  string repository = 2;
+  string reference = 3;
+
+  // If the image should exclude depdendencies. When set to true, a incomplete
+  // image without descriptors for dependencies of the requested types is
+  // returned.
+  bool exclude_dependencies = 4;
```

Comment:
Do we need to start with this option? I understand the distinction and potential use case, but when would we need to return an `Image` containing the `pet.v1.Pet` type without the `google.type.DateTime` definition?

For request builder and other use cases, I'd imagine that we still need to return the tight transitive closure around the type, so that all of the message's field could be referenced and included in an auto-complete placeholder.

To be clear, I'm referring to [this](https://buf.build/acme/petapis/file/main/pet/v1/pet.proto#L22) where the `google.type.DateTime` fields are unclear from the `Pet` type on its own:

```protobuf
// Pet represents a pet in the pet store.
message Pet {
  PetType pet_type = 1;
  string pet_id = 2;
  string name = 3;
  google.type.DateTime created_at = 4;
}
```

Reply:
Separate point that I now notice isn't clear, with dependencies here I mean to refer to imported files (`repeated string dependencies` in the FileDescriptor definitions. Below with `match_dependencies` I had meant to refer to imports at the buf module level, not at the file level.

------

The use case I was eyeing for this option is that if I just want a list of all services (and their methods), I don't want the request's dependencies yet at this point. Say this is part of a large module:
```
message DoRequest {
    google.type.DateTime created_at = 4;
}
service PetAPI {
  rpc Do(DoRequest) returns (...);
}
```
And I would like filedescriptors that contain requests, but don't want things they refer to. For a large module like googleapis, querying for only the services while not receiving all the enums and messages that are required for making calls with them seems useful (maybe I should dig up some numbers)?

I don't think this is the common case though, which is why I made exclude (and false by default).